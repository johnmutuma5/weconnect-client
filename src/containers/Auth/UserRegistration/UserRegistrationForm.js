import React from 'react';
import { withRouter } from 'react-router-dom';
import Form, {formProcessComplete, formProcessFailed, formInput} from '../../../components/UI/utils/Form';
import {createAccount} from '../../../store/resources/auth';
import Modal from '../../../components/UI/Modal/Modal';
import Prompt from '../Prompt/Prompt';

// import './UserRegistrationForm.css';

export class UserRegistrationForm extends React.Component {

    constructor(props) {
        super();
        this.state = {
            values: {
                first_name: '',
                last_name: '',
                username: '',
                password: '',
                mobile: '',
                email: '',
                gender: ''
            }
        }
    }

    onSubmit(e) {
        e.preventDefault()
        console.log(this.state.values)
        createAccount(this.state.values)
            .then((response) => {
                const msg = response.msg;
                const next = '/businesses';
                this.setState(formProcessComplete(msg), this.redirect(next));
            })
            .catch((error) => {
                const msg = error.msg;
                this.setState(formProcessFailed(msg));
            });
    }

    redirect(path) {
        // returns a timed-out callback function for setState
        return () => {
            setTimeout(()=>this.props.handleReadyToLogin() ,1500);
        }
    }

    render() {
        const modalTitle = 'Register a user account';
        let form = <Form
                        formContext = { this }
                        loadElements = { this.loadElements.bind(this) }
                        onSubmit = { this.onSubmit.bind(this) } />
        return (
            <Modal title={ modalTitle } isVisible={this.props.isVisible}>
                <Prompt
                    ask={'Already have an account?'}
                    handleClicked={ this.props.handleReadyToLogin} >
                    { 'Login' }</Prompt>
                { form }
            </Modal>
        );
    }

    loadElements() {
        return [
            {
                elementType: 'fieldset',
                key: 'usrreg1',
                children: [
                    {
                        elementType: 'legend',
                        children: 'Basic and Contact Data'
                    },
                    formInput('text', 'first_name', this.state.values.first_name, 'First Name'),
                    formInput('text', 'last_name', this.state.values.last_name, 'Last Name'),
                    // formInput('text', 'gender', this.state.values.gender, 'Gender'),
                    {
                        elementType: 'select',
                        attributes: {
                            name: 'gender'
                        },
                        children: [
                            {
                                elementType: 'option',
                                attributes: {
                                    value: 'Male'
                                },
                                children: 'Male'
                            },
                            {
                                elementType: 'option',
                                attributes: {
                                    value: 'Female'
                                },
                                children: 'Female'
                            }
                        ]
                    },
                    formInput('email', 'email', this.state.values.email, 'Email'),
                    formInput('text', 'mobile', this.state.values.mobile, '254700000000')
                ]
            },
            {
                elementType: 'fieldset',
                key: 'usrreg2',
                children: [
                    {
                        elementType: 'legend',
                        children: 'Login Details'
                    },
                    formInput('text', 'username', this.state.values.username, 'Username'),
                    formInput('password', 'password', this.state.values.password, 'Password')
                ]
            },
            {
                elementType: 'fieldset',
                key: 'usrreg3',
                children: [
                    formInput('submit', 'submit', 'Submit', undefined)
                ]
            }
        ]
    }
}

export default withRouter(UserRegistrationForm);
