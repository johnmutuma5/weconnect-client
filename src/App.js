import React, { Component } from 'react';
import Layout from './containers/Layout/Layout';
import { Route } from 'react-router-dom';
import { Switch } from 'react-router';
import Businesses from './containers/Businesses/Businesses';
import BusinessesFilter from './containers/BusinessesFilter/BusinessesFilter';
import SearchBusinesses from './containers/Search/Search';
import UserRegistrationForm from './containers/Auth/UserRegistration/UserRegistrationForm';
import BusinessProfile from './containers/BusinessProfile/BusinessProfile';
import ResourcesProvider, { createStore } from './store';
import weConnectReducer from './store/baseReducer';


export const store = createStore(weConnectReducer);

class App extends Component {
  render() {
    return (
        <ResourcesProvider store={ store }>
            <Layout>
                <Switch>
                    <Route exact path="/" component={ Businesses } />
                    <Route exact path="/businesses" component={ Businesses } />
                    <Route exact path="/businesses/filter" component={ BusinessesFilter } />
                    <Route exact path="/businesses/search" component={ SearchBusinesses } />
                    <Route exact path="/businesses/:id" component={ BusinessProfile } />
                </Switch>
            </Layout>
        </ResourcesProvider>
    );
  }
}

export default App;
