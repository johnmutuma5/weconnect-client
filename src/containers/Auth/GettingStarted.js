import React from 'react';
import PropTypes from 'prop-types';
import UserRegistrationForm from './UserRegistration/UserRegistrationForm';
import UserLoginForm from './UserLogin/UserLoginForm';
import { toggleGettingStarted } from '../../store/actions/actions';
import Button from '../../components/UI/Button/Button';
import Backdrop from '../../components/UI/Backdrop/Backdrop';
import Aux from '../../hoc/Aux';

class GettingStarted extends React.Component {

    constructor(props, context) {
        super();
        this.state = {
            startBy: 'signUp'
        }
    }

    render() {
        const store = this.context.store;
        let form = <UserRegistrationForm
                        isVisible={ this.props.visible }
                        handleReadyToLogin={ this.handleReadyToLogin.bind(this) }/>
        // switch form between sign up, login and reset password
        form = this.switchForm(form);
        return (
            <Aux>
                { form }
                <Backdrop
                    active={ this.props.visible }
                    id={ 'GettingStartedBackdrop' }
                    click={ () => store.dispatch(toggleGettingStarted()) }/>
            </Aux>
        );
    }


    handleReadyToLogin(e) {
        this.setState({startBy: 'login'});
    }

    handleReadyToSignUp(e) {
        this.setState({startBy: 'signUp'});
    }

    switchForm(form) {
        switch (this.state.startBy) {
            case 'login':
                form = <UserLoginForm
                            isVisible={ this.props.visible }
                            handleReadyToSignUp={ this.handleReadyToSignUp.bind(this) }/>;
                break;
            case 'resetPassword':
                form = 'resetPassword'; // to add this form
                break;
            default:

        }
        return form;
    }
}

export const GettingStartedButton = (props, context) => {
    const store = context.store;
    const click = () => store.dispatch(toggleGettingStarted());
    return (
        <Button id={props.id}
                type='dark'
                onClickHandler={ click }>
                    { 'Get Started' }
        </Button>
    )
}

GettingStarted.contextTypes = { store: PropTypes.object.isRequired }
GettingStartedButton.contextTypes = { store: PropTypes.object.isRequired }


export default GettingStarted;
