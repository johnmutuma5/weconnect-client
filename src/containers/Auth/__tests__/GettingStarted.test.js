import GettingStarted from '../GettingStarted';
import UserRegistrationForm from '../UserRegistration/UserRegistrationForm';
import { store } from '../../../App';

describe('<GettingStarted />', () => {
    let context;

    beforeEach(() => {
        context = {
            context: { store },
            childContextTypes: {
                store: PropTypes.object
            }
        }
    });


    it('renders correctly', () => {
        const wrapper = shallow(<GettingStarted />, context);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders user login form as the default form', () => {
        expect.assertions(2);
        const wrapper = mount(<MemoryRouter><GettingStarted /></MemoryRouter>, context);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.text()).toContain('Login');
    });

    it('toggles sign up and login form with button click', () => {
        expect.assertions(2);
        const wrapper = mount(<MemoryRouter>
                                <GettingStarted />
                              </MemoryRouter>, context);
        wrapper.find('.Button').simulate('click');
        const state = wrapper.find(GettingStarted).instance().state;
        expect(state).toEqual({startBy: 'signUp'});
        expect(wrapper.text()).toContain('Register a user account');
    });
});
