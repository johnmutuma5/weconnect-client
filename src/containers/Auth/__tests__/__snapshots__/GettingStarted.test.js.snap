// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GettingStarted /> renders correctly 1`] = `
<withRouter(UserRegistrationForm)
  handleReadyToLogin={[Function]}
/>
`;

exports[`<GettingStarted /> renders user sign up form as the default form 1`] = `
<div
  className="Modal invisible"
>
  <p>
    Register a user account
  </p>
  <span />
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
        "margin": "15px 0",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
        }
      }
    >
      Already have an account?
    </div>
    <div
      className="Button dark"
      onClick={[Function]}
    >
       
      Login
    </div>
  </div>
  <article
    className="form_container"
  >
    <form
      onSubmit={[Function]}
    >
      <fieldset
        key="usrreg1"
      >
        <legend
          key="Basic and Contact Data"
        >
          Basic and Contact Data
        </legend>
        <input
          key="first_name"
          name="first_name"
          onChange={[Function]}
          placeholder="First Name"
          type="text"
          value=""
        />
        <input
          key="last_name"
          name="last_name"
          onChange={[Function]}
          placeholder="Last Name"
          type="text"
          value=""
        />
        <select
          name="gender"
          onChange={[Function]}
        >
          <option
            disabled={true}
            hidden={true}
            selected={true}
            value=""
          >
            Select Gender...
          </option>
          <option
            value="Male"
          >
            Male
          </option>
          <option
            value="Female"
          >
            Female
          </option>
        </select>
        <input
          key="email"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          type="email"
          value=""
        />
        <input
          key="mobile"
          name="mobile"
          onChange={[Function]}
          placeholder="254700000000"
          type="text"
          value=""
        />
      </fieldset>
      <fieldset
        key="usrreg2"
      >
        <legend
          key="Login Details"
        >
          Login Details
        </legend>
        <input
          key="username"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          type="text"
          value=""
        />
        <input
          key="password"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </fieldset>
      <fieldset
        key="usrreg3"
      >
        <input
          key="submit"
          name="submit"
          onChange={[Function]}
          type="submit"
          value="Submit"
        />
      </fieldset>
    </form>
  </article>
</div>
`;
